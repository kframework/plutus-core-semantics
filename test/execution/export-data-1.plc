(program

(module SomeMod (import) (export ((MyType (This NotThis))) ())
  (data MyType ()
        (This (integer))
        (NotThis))
)

(module Foo (import SomeMod) (export () ())
  (define result
    (case (con SomeMod.This 1)
          (SomeMod.This (x)  (builtin addInteger 10 x))
          (SomeMod.NotThis ()  -1))))

)
