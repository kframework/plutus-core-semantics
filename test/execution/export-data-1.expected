<T>
  <k> . </k>
  <lenv> .Map </lenv>
  <genv> ( Foo . result ) |-> 11 </genv>
  <modules>
    <module>
      <curr> 0 </curr>
      <name> Foo </name>
      <data> .TyExpList </data>
      <decl> .NameList </decl>
      <import> SomeMod  .ModNmList </import>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </module>
    <module>
      <curr> 0 </curr>
      <name> SomeMod </name>
      <data> ( MyType ( This  ( NotThis  .ConList ) ) )  .TyExpList </data>
      <decl> .NameList </decl>
      <import> .ModNmList </import>
      <exportTy> ( MyType ( This  ( NotThis  .ConList ) ) )  .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </module>
  </modules>
</T>
